QualityControl/test/public/pages/layout-view.test.js

```diff
@@ -10,153 +10,223 @@
+    { timeout },
+    async () => {
+      const buttonPath = 'header > div > div:nth-child(3) > div > button:nth-child(1)';
+      const duplicateButton = await page.evaluate((buttonPath) => document.querySelector(buttonPath).title, buttonPath);
+      strictEqual(duplicateButton, 'Duplicate layout');
+    },
+  );
+
+  await testParent.test(
+    'should have one delete button in the header to delete layout',
+    { timeout },
+    async () => {
+      const buttonPath = 'header > div > div:nth-child(3) > div > button:nth-child(4)';
+      const deleteButton = await page.evaluate((buttonPath) => document.querySelector(buttonPath).title, buttonPath);
+      strictEqual(deleteButton, 'Delete layout');
+    },
+  );
+
+  await testParent.test(
+    'should have one link button in the header to download layout skeleton',
+    { timeout },
+    async () => {
+      const buttonPath = 'header > div > div:nth-child(3) > div > a';
+      const editButton = await page.evaluate((buttonPath) => document.querySelector(buttonPath).title, buttonPath);
+      strictEqual(editButton, 'Export layout skeleton as JSON file');
+    },
+  );
+
+  await testParent.test(
+    'should have one edit button in the header to go in edit mode',
+    { timeout },
+    async () => {
+      const buttonPath = 'header > div > div:nth-child(3) > div > button:nth-child(3)';
+      const editButton = await page.evaluate((buttonPath) => document.querySelector(buttonPath).title, buttonPath);
+      strictEqual(editButton, 'Edit layout');
+    },
+  );
+
+  await testParent.test(
+    'should click the edit button in the header and enter edit mode',
+    { timeout },
+    async () => {
+      const editButtonPath = 'header > div > div:nth-child(3) > div > button:nth-child(3)';
+      await page.locator(editButtonPath).click();
+    },
+  );
+
+  await testParent.test(
+    'should have input field for changing layout name in edit mode',
+    { timeout },
+    async () => {
+      const inputPath = 'header > div > div:nth-child(3) > input';
+      await page.evaluate((inputPath) => document.querySelector(inputPath), inputPath);
+    },
+  );
+
+  await testParent.test(
+    'should have number input field for allowing users to change auto-tab value',
+    { timeout },
+    async () => {
+      await page.waitForSelector('#inputDescription', { timeout: 5000 });
+    },
+  );
+
+  await testParent.test(
+    'should have a tree sidebar in edit mode',
+    { timeout },
+    async () => {
+      const secondElementPath = 'nav table tbody tr:nth-child(2)';
+      await page.locator(secondElementPath).click();
+      const rowsCount = await page.evaluate((secondElementPath) =>
+        document.querySelectorAll(secondElementPath).length, secondElementPath);
+      strictEqual(rowsCount, 1);
+    },
+  );
+
+  await testParent.test(
+    'should have filtered results on input search filled',
+    { timeout },
+    async (t) => {
+      t.skip('Skip test execution for now');
```

Comment:
Is there a reason for which we skip this test?

Reply:
Yes, I asked you about this. We should discuss it because it's not working in production.

Reply:
Understood. I created ticket: OGUI-1568 for this issue. Could you please add in the description some steps on how to replicate this bug in production? Thank you!